plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/securepassword/PasswordHasher.java

```diff
@@ -36,7 +36,7 @@ public static boolean verifyPWHash(char[] pwd, java.lang.String pwdhash) throws
 		byte[] res = null;
 		int keysize = 160;
 
-		CrySLCodeGenerator.getInstance().includeClass("java.security.PBEKeySpec").addParameter(pwd, "password").addParameter(salt, "salt").addParameter(keysize, "keylength")
+		CrySLCodeGenerator.getInstance().includeClass("java.security.SecureRandom").addParameter(salt, "next").includeClass("java.security.PBEKeySpec").addParameter(pwd, "password").addParameter(salt, "salt").addParameter(keysize, "keylength")

```

Comment:
This is incorrect, though. The verification must not randomize the hash, but take the one that is used for the creation of the hash (as defined in line 35).